src/cache/redis.rs

```diff
@@ -1,95 +1,99 @@
 use crate::cache::Cache;
 use crate::config::{redis_scan_count, redis_uri};
-use r2d2::{Pool, PooledConnection};
-use redis::{self, pipe, Commands, FromRedisValue, Iter, ToRedisArgs};
-
-type RedisPool = Pool<redis::Client>;
-type RedisConnection = PooledConnection<redis::Client>;
+use bb8_redis::bb8::{self, Pool, PooledConnection};
+use bb8_redis::redis::{cmd, AsyncCommands, AsyncIter, Cmd, Pipeline, ToRedisArgs};
+use bb8_redis::RedisConnectionManager;
+
+type RedisPool = Pool<RedisConnectionManager>;
+type RedisConnection<'a> = PooledConnection<'a, RedisConnectionManager>;
+
+async fn create_pool() -> RedisPool {
+    let manager = RedisConnectionManager::new(redis_uri())
+        .expect("Establishing connection with redis instance failed");
+    bb8::Pool::builder()
+        .max_size(15) // default is 10
+        .build(manager) // we can technically also set a connection await timeout if necessary
```

Comment:
nitpick: the comment here is unrelated with the `.build()` right?

Reply:
No strong opinion on this one. It's more the defaults and the API itself that can change and I think the best way to see it is by opening the docs 😄